spyderlib/restart_app.py

```diff
@@ -128,19 +186,52 @@ def main():
     # Adjust the command and/or arguments to subprocess depending on the OS
     shell = os.name != 'nt'
 
-    # Wait for original process to end before launching the new instance
-    while True:
+    # Before launching a new Spyder instance we need to make sure that the
+    # previous one has closed. We wait for a fixed and "reasonable" amount of
+    # time and check, otherwise an error is launched
+    wait_time = 60*5  # Seconds

```

Comment:
So we're waiting for 5 min. Isn't that way too much time? I mean, specially if we are not giving our users any visual feedback about what's happening :-)

Two questions:
1. Isn't 2 or 3 min at most enough?
2. Can't we show a throbber or something while doing this? I mean, something like we already have when creating kernels for our IPython consoles?